src/components/TextInput/style.scss

```diff
@@ -2,109 +2,79 @@
 
 .wrapper {
   @extend %input--wrapper;
+  @extend %type--body;
+
+  box-sizing: border-box;
   display: flex;
   flex-direction: row;
+  font-size: $typeBase;
+  justify-content: stretch;
+
   &:hover {
     @extend %input--wrapper--hover;
-    .fix {
-      border-top: 1px solid $cGray200;
-      border-right: 1px solid $cGray200;
-      border-bottom: 1px solid $cGray200;
-      border-left: 1px solid $cGray200;
+
+    .prefix,
+    .suffix {
+      border-color: $cGray200;
     }
   }
-}
 
-.wrapperLarge {
-  @extend %input--wrapper--large;
-  .fix {
-    padding: 0 15px;
-  }
-}
+  &:focus-within {
+    @extend %input--wrapper--focus;
 
-.wrapperFocused {
-  @extend %input--wrapper--focus;
-  .fix {
-    border: 1px solid $cGray200;
+    .prefix,
+    .suffix {
+      border-color: $cGray200;
+    }
   }
-}
 
-.wrapperDisabled {
-  @extend %input--wrapper--disabled;
-  .fix {
-    color: $cGray300;
-    border: 1px solid $cGray400;
-  }
-  &:hover {
-    .fix {
-      border: 1px solid $cGray400;
+  &.wrapperDisabled {
+    @extend %input--wrapper--disabled;
+
+    .prefix,
+    .suffix {
+      border-color: $cGray400;
     }
   }
-}
 
-.wrapperError {
-  @extend %input--wrapper--error;
-  &:hover {
+  &.wrapperError,
+  &.wrapperError:hover {
     @extend %input--wrapper--error;
-  }
-  .fix {
-    border: 1px solid $cWarningRed;
-  }
-  &:hover {
-    .fix {
-      border: 1px solid $cWarningRed;
+
+    .prefix,
+    .suffix {
+      border-color: $cWarningRed;
     }
   }
 }
 
 .input {
   @extend %input;
   @extend %type--body;
+
   border: 0;
+  flex: 1 1 auto;
   outline: none;
-  box-sizing: border-box;
-  width: 100%;
-  height: 100%;
-  order: 1;
-}
-
-.large {
-  @extend %input--large;
-}
 
-.disabled {
-  @extend %input--disabled;
+  &:disabled {
+    @extend %input--disabled;
+  }
 }
 
-.fix {
-  @extend %fontFamily--body;
+.prefix,
+.suffix {
+  border: 0 solid $cGray300;
```

Comment:
is this empty border needed?

Reply:
I'm actually just setting `border-width` and `border-color`, but IDEs sometimes complain that having those separately can be optimized.

If you'll look, I later set `border-right/left` for `.suffix/prefix`.